### PR TITLE
Changes to number string definitions

### PR DESCRIPTION
**Pull request by MichaelClerx**
Normative changes:

- "European numerals" are now called "Digits" #178
- Non-negative integer string must be one or more digits: Not sure if we need to allow a leading plus here? I've gone for "no" in this PR.
- Integer strings start with + or -, then one or more digits
- made clear at the outset that the exponent in real number strings is _optional_ #183 
- tweaked the wording so that "" is not a number string, nor is ".", but "1." and ".1" both are

Informative changes:

- Numerals --> digits #33
- Basic latin --> Basic latin alphabetic characters ("Basic Latin" includes digits, @ etc., all of ascii! https://en.wikipedia.org/wiki/Basic_Latin_(Unicode_block) )
- small tweaks

### CODE REVIEW

**Comment by MichaelClerx** [comment]
Closes #199

**Comment by MichaelClerx** [comment]
Closes #183

File: src/reference/informative/informA1.rst [hunk 1]
```diff
@@ -16,35 +16,39 @@
...
-    It contains the upper and lowercase alphabet without decoration:
+    The codes themselves are arranged into blocks representing sets of characters, and the Basic Latin block is one of these.
+    It contains, amongst other things, the upper and lowercase alphabet without decoration:
 
       :code:`abcdefghijklmnopqrstuvwxyz` (symbols between :unicode:`0061` and :unicode:`007A`)
       :code:`ABCDEFGHIJKLMNOPQRSTUVWXYZ` (symbols between :unicode:`0041` and :unicode:`005A`)
 
-    The European numerals are the Unicode set:
+    These characters are refferd to in CellML as the Basic Latin alphabetic characters.
+
```
Comment (kerimoyle) [inline src/reference/informative/informA1.rst:33]: Do we want quotes or italics for "Basic Latin alphabetic characters"?

File: src/reference/informative/informA1.rst [hunk 2]
```diff
@@ -16,35 +16,39 @@
...
+    The digits are the Unicode characters:
 
       :code:`0123456789` (symbols between :unicode:`0030` and :unicode:`0039`)
 
+    Together with the Basic Latin alphabetic characters, they form the Basic Latin alphanumerical characters.
```
Comment (kerimoyle) [inline src/reference/informative/informA1.rst:38]: Is it "alphabetic" and "alphanumeric"  ... or "alphabetical" and "alphanumerical"? There doesn't seem to be any difference in the meaning of each of the pairs.

File: src/reference/informative/informA1.rst [hunk 3]
```diff
@@ -16,35 +16,39 @@
...
-    - the tab (:unicode:`0009`), 
+
+    - the space (:unicode:`0020`),
+    - the tab (:unicode:`0009`),
     - the carriage return (:unicode:`000D`), and
-    - the line feed (:unicode:`000A`).  
+    - the line feed (:unicode:`000A`).
 
-    Together these symbols define the only characters which can be interpreted into values for CellML element attributes and content. 
+    Together these symbols define the only characters which can be interpreted into values for CellML names and numbers.
```
Comment (kerimoyle) [inline src/reference/informative/informA1.rst:54]: It's not quite the same thing?  We can only use these characters anywhere that something needs to be interpreted, including references etc?  Hmmm ... and now thinking about it, what about all the possible symbols in URLs?? We haven't listed them anywhere ...?

File: src/reference/sectionA_definitions.inc [hunk 4]
```diff
@@ -250,15 +250,15 @@ The following data representation formats are defined for use in this specificat
 
    .. container:: issue-data-repr-basic-real-sign
 
-      2. SHALL, when the basic real number being represented is negative, begin with the Basic Latin hyphen-minus character :unicode:`002D` as the sign indicator.
+      2. MAY begin with a single Basic Latin plus character :unicode:`002B` or a Basic Latin hyphen-minus character :unicode:`002D` as the sign indicator.
```
Comment (kerimoyle) [inline src/reference/sectionA_definitions.inc:253]: When it's negative it must begin with the - sign?  

- MAY, when the basic real number being represented is positive, begin with the Basic Latin plus character :unicode:`whateveritis` as the sign indicator.
- SHALL, when the basic real number being represented is negative, begin with the Basic Latin hyphen-minus character :unicode:`002D` as the sign indicator.

**Comment by kerimoyle** [comment]
Looks pretty good, thanks :) Couple of follow up questions:
- Did you do a global search for the old `specA_european_numeral` link so we're sure it's not referenced elsewhere?  
- Also, it's probably worth checking that any manually added numbers haven't been broken?  Just search for the ":hardcodedref:" tag to see where we've used section numbers in the text.

**Comment by kerimoyle** [comment]
Also would be good to check that none of the informative examples are broken by this change?  Might add that to my list though ...

File: src/reference/informative/informA1.rst (hunk 1 above)
Comment (MichaelClerx) [reply src/reference/informative/informA1.rst:33]: refferd --> referred

File: src/reference/sectionA_definitions.inc [hunk 5]
```diff
@@ -218,7 +218,7 @@ The following data representation formats are defined for use in this specificat
 
    .. container:: issue-data-repr-nneg-int-euro-num
 
-      2. SHALL consist entirely of :ref:`European numerals<specA_european_numeral>`.
+      2. SHALL consist of one or more :ref:`digits<specA_digit>`.
```
Comment (MichaelClerx) [inline src/reference/sectionA_definitions.inc:221]: The "non-negative integer string" can go, not used in the spec anymore

File: src/reference/informative/informA1.rst (hunk 3 above)
Comment (MichaelClerx) [reply src/reference/informative/informA1.rst:54]: We refer to the xlink spec for URLs

> The value of this attribute SHALL be a valid locator href, as defined in Section 5.4 of the XLink specification.
> The href attribute SHALL be treated according to the XLink specification, by applying the rules for simple-type elements.

Comment (MichaelClerx) [reply src/reference/informative/informA1.rst]: Have replaced this sentence by something less restrictive. (Just saying these classes will be uesd, not that they will be used exclusively!)

File: src/reference/sectionA_definitions.inc (hunk 4 above)
Comment (MichaelClerx) [reply src/reference/sectionA_definitions.inc:253]: I'm fine with having positive and negative numbers defined outside the spec, in the usual way. It's captured by the first rule (base 10 int representation) I think

**Comment by MichaelClerx** [comment]
> Looks pretty good, thanks :) Couple of follow up questions:
> 
>     * Did you do a global search for the old `specA_european_numeral` link so we're sure it's not referenced elsewhere?

Yes

**Comment by MichaelClerx** [comment]
> 
>     * Also, it's probably worth checking that any manually added numbers haven't been broken?  Just search for the ":hardcodedref:" tag to see where we've used section numbers in the text.

the result of grepping is

> ./src/conf.py:# more easily by searching for the "hardcodedref" tag.
> ./src/conf.py:.. role:: hardcodedref 

so maybe they're all in PRs ?

**Comment by MichaelClerx** [comment]
Fixed the informative spec now too

**Comment by MichaelClerx** [comment]
Happy to merge now @kerimoyle ?

**Comment by kerimoyle** [comment]
> so maybe they're all in PRs ?

Could be ... though I'm a bit surprised.  Is your branch up to date?

**Comment by MichaelClerx** [comment]
> Are we changing "alphabetic" into "alphabetical" and "alphanumeric" into "alphanumerical" in this PR or somewhere else?

Separate PR

**Comment by MichaelClerx** [comment]
> > so maybe they're all in PRs ?
> 
> Could be ... though I'm a bit surprised. Is your branch up to date?

Ah. Done it better now. Checked and they're ok!

**Pull request merged**